tests/unit/dialog/dialog_core.js

```diff
@@ -7,6 +7,37 @@
 // TODO add teardown callback to remove dialogs
 module("dialog: core");
 
+test( "markup structure", function() {
+	expect( 13 );
+	var element = $( "<div></div>" ).dialog({
+			buttons: [ {
+				text: "Ok",
+				click: $.noop
+			} ]
+		}),
+		widget = element.dialog( "widget" ),
+		titlebar = widget.find( ".ui-dialog-titlebar" ),
+		close = titlebar.find( ".ui-button" ),
+		buttonpane = widget.find( ".ui-dialog-buttonpane" );
+
+	ok( widget.hasClass( "ui-dialog" ), "Widget has class ui-dialog" );
+	ok( widget.hasClass( "ui-widget" ), "Widget has class ui-widget" );
+	ok( widget.hasClass( "ui-widget-content" ), "Widget has class ui-widget-content" );
+	ok( widget.hasClass( "ui-dialog-buttons" ), "Widget has class ui-dialog-buttons" );
+	ok( titlebar.hasClass( "ui-dialog-titlebar" ), "Titlebar has class ui-dialog-titlebar" );
+	ok( titlebar.hasClass( "ui-widget-header" ), "Titlebar has class ui-widget-header" );
+	ok( titlebar.find( ".ui-dialog-title" ).length, "Title hasClass ui-dialog-title" );
```

Comment:
I'm not very familiar at all with how custom asserts work for how exactly this would look, but it seems like if we were going to do something like this wouldn't it make more sense to have the second arg be either an array or space delimited string of classes? That way you could call it once per elements?